core/app/models/spree/store_credit.rb

```diff
@@ -23,10 +23,12 @@ class Spree::StoreCredit < Spree::PaymentSource
   validates_numericality_of :amount_used, { greater_than_or_equal_to: 0 }
   validate :amount_used_less_than_or_equal_to_amount
   validate :amount_authorized_less_than_or_equal_to_amount
+  validates :expires_at, presence: true, unless: ->(credit) { credit.category.non_expiring? }
```

Comment:
@luukveenis - yes. It also means the current implementation is far more complicated than it should be (to achieve the same thing). I'm personally new to Solidus eCom but seems like it is a really difficult affair to backtrack on design decisions, due to fear that it will break someone's proprietary setup?